src/kOS/Execution/Stack.cs

```diff
@@ -48,9 +52,19 @@ private void ThrowIfInvalid(object item)
         /// <returns></returns>
         private object ProcessItem(object item)
         {
-            if (item is float)
+            if (item is float) {
                 // promote floats to doubles
-                return Convert.ToDouble(item);
+                item = Convert.ToDouble (item);
+            }
+            if (item is double)
+            {
+                if (!Double.IsNaN((double)item) && Int32.MinValue < (double)item && (double)item < Int32.MaxValue)
+                {
+                    int intPart = Convert.ToInt32(item);
+                    if ((double)item == (double)intPart)
+                        item = intPart;
+                }
+            }
```

Comment:
The current version of kOS is doing the same check and conversion of round doubles to int in the Calculator class: https://github.com/KSP-KOS/KOS/blob/develop/src/kOS.Safe/Compilation/Calculator.cs#L171. I tested it: `switch to 2.0.` doesn't work without my fix but `switch to 2.0 + 0.0.` does.